src/terminal/terminal.c

```diff
@@ -335,6 +335,68 @@ guac_terminal_options* guac_terminal_options_create(
     return options;
 }
 
+/**
+ * Calculate the available space within the terminal 
+ * and store the results in the pointer arguments.
+ *
+ * @param terminal
+ *     The terminal provides character width and height for calculations.
+ *     This function requires outer_width, outer_height, 
+ *     display->margin, display->char_height, and display->char_width
+ *     to be set on terminal in order to calculate available dimensions.
+ * 
+ * @param aw
+ *     Set with the available width for text of the terminal, in pixels.
+ * 
+ * @param ah
+ *     Set with the available height for text of the terminal, in pixels.
+ * 
+ * @param columns
+ *     Set with the available width for text of the terminal, by column count.
+ * 
+ * @param rows
+ *     Set with the available height for text of the terminal, by row count.
+ */
+static void get_available_dimensions(guac_terminal* term,
```

Comment:
At this point, having the function called `get_...` is pretty misleading - usually you'd expect a getter to return a value, which this doesn't do. 

Maybe `calculate_and_save_...` or similar might be better?

Reply:
Thanks! That's a lot simpler - though now that the function updates the terminal struct directly, the block comment for the function should actually say that - reading the comment as it exists now, you'd think it only updates the passed-in pointers.